File: archived/DateTimeFormatting/js/js/scenario6-timeZone.js
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";
    var page = WinJS.UI.Pages.define("/html/scenario6-timeZone.html", {
        ready: function (element, options) {
            document.getElementById("displayButton").addEventListener("click", doDisplay, false);
        }
    });

    function doDisplay() {
        // This scenario illustrates time zone support in DateTimeFormatter class
        SdkSample.clearFormattingResults();

        // Additional time zones we will show.
        var timeZones = ["UTC", "America/New_York", "Asia/Kolkata"];
        
        // Create formatter object using longdate and longtime template
        var formatter = new Windows.Globalization.DateTimeFormatting.DateTimeFormatter("longdate longtime");
        
        // Obtain the date that will be formatted.
        var dateToFormat = new Date();
        
        // Show current time.
        var results = "Current date and time\n";
        results += "In Local time zone: " + formatter.format(dateToFormat) + "\n";
        timeZones.forEach(function (timeZone) {
            results += "In " + timeZone + " time zone: " + formatter.format(dateToFormat, timeZone) + "\n";
        });
        results += "\n";

        // Show a time on 14th day of second month of next year.
        // (Note that month numbers start at zero in JavaScript, so month 1 is the second month.)
        // Note the effect of daylight saving on the results.
        results += "Same time on 14th day of second month of next year\n";
        dateToFormat = new Date(dateToFormat.getFullYear() + 1, 1, 14, dateToFormat.getHours(), dateToFormat.getMinutes(), dateToFormat.getSeconds());
        results += "In local time zone: " + formatter.format(dateToFormat) + "\n";
        timeZones.forEach(function (timeZone) {
            results += "In " + timeZone + " time zone: " + formatter.format(dateToFormat, timeZone) + "\n";
        });
        results += "\n";

        // Show a time on 14th day of tenth month of next year.
        // Note the effect of daylight saving on the results.
        results += "Same time on 14th day of tenth month of next year\n";
        dateToFormat.setMonth(9);
        results += "In local time zone: " + formatter.format(dateToFormat) + "\n";
        timeZones.forEach(function (timeZone) {
            results += "In " + timeZone + " time zone: " + formatter.format(dateToFormat, timeZone) + "\n";
        });
        results += "\n";

        // Display the results.
        document.getElementById("output").innerText = results;
    }
})();

File: Samples/DateTimeFormatting/cppwinrt/Scenario6_TimeZone.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario6_TimeZone.h"
#include "Scenario6_TimeZone.g.cpp"

using namespace winrt;
using namespace Windows::Foundation;
using namespace Windows::Globalization;
using namespace Windows::Globalization::DateTimeFormatting;
using namespace Windows::UI::Xaml;

namespace
{
    void GenerateResult(std::wostream& results, DateTimeFormatter const& formatter, PCWSTR timeZone, DateTime dateToFormat)
    {
        // Perform the actual formatting. 
        results << L"In " << timeZone << L" time zone: " <<
            std::wstring_view(formatter.Format(dateToFormat, timeZone)) << std::endl;
    }
}

namespace winrt::SDKTemplate::implementation
{
    Scenario6_TimeZone::Scenario6_TimeZone()
    {
        InitializeComponent();
    }

    void Scenario6_TimeZone::Display_Click(IInspectable const&, RoutedEventArgs const&)
    {
        // This scenario illustrates time zone support in DateTimeFormatter class

        // Displayed time zones (other than local time zone)
        std::array timeZones{ L"UTC", L"America/New_York", L"Asia/Kolkata" };

        // Keep the results here
        std::wostringstream results;

        // Create formatter object using longdate and longtime template
        DateTimeFormatter dateFormatter(L"longdate longtime");

        // Create date/time to format and display.
        Calendar calendar;
        DateTime dateToFormat = calendar.GetDateTime();

        // Show current time.
        results << L"Current date and time" << std::endl <<
            L"In Local time zone: " << std::wstring_view(dateFormatter.Format(dateToFormat)) << std::endl;
        for (auto&& timeZone : timeZones)
        {
            GenerateResult(results, dateFormatter, timeZone, dateToFormat);
        }
        results << std::endl;

        // Show a time on 14th day of second month of next year.
        // Note the effect of daylight saving on the results.
        calendar.AddYears(1); calendar.Month(2); calendar.Day(14);
        dateToFormat = calendar.GetDateTime();
        results << L"Same time on 14th day of second month of next year" << std::endl <<
            L"In Local time zone: " << std::wstring_view(dateFormatter.Format(dateToFormat)) << std::endl;
        for (auto timeZone : timeZones)
        {
            GenerateResult(results, dateFormatter, timeZone, dateToFormat);
        }
        results << std::endl;

        // Show a time on 14th day of tenth month of next year.
        // Note the effect of daylight saving on the results.
        calendar.AddMonths(8);
        dateToFormat = calendar.GetDateTime();
        results << L"Same time on 14th day of tenth month of next year" << std::endl <<
            L"In Local time zone: " << std::wstring_view(dateFormatter.Format(dateToFormat)) << std::endl;
        for (auto timeZone : timeZones)
        {
            GenerateResult(results, dateFormatter, timeZone, dateToFormat);
        }
        results << std::endl;

        // Display the results.
        OutputTextBlock().Text(results.str());
    }
}
